pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,373 @@
+            var total = tasks.length;
+
+            function step(qtasks) {
+                var e = qtasks.shift();
+                if (!e) {
+                    deferred.resolve();
+                    return;
+                }
+                var task_name = e[0];
+                var args = e[1];
+                var task = e[2];
+
+                task.apply(null, args).done(function(response) {
+                    deferred.notify("created", response, qtasks.length - total, total);
+                    step(qtasks);
+                }).fail(function(ex, response) {
+                    if (ex.status == 409) {
+                        deferred.notify("skipped", response, qtasks.length - total, total);
+                        step(qtasks);
+                    } else {
+                        deferred.reject(ex, response);
+                    }
+                });
+            }
+
+            step(tasks);
+
+            return deferred.promise();
+        }
+
+        function show_progress_message(message) {
+            is_deploying.show();
+            $('#deploy-update-message').text(message);
+        }
+
+        function hide_progress_message() {
+            is_deploying.hide();
+        }
+
+        function remove_notifications() {
+            $('div.container-fluid.alert').remove();
+            deploy_dialog_remove_errors();
+        }
+
+
+        function action_in_progress() {
+            return (is_deploying.is(':visible'));
+        }
+
+        return {
+            'deploy_app': deploy_app
+        };
+
+    }
+
+    function deploy_app() {
+        //alert("deploy_app")
+        var jsondata = "";
+        deploy_dialog_remove_errors();
+        jsondata = appdeployer.jsondata;
+
+        var ns = $('#deploy-app-namespace-field').val();
+        if ($('#deploy-app-namespace-field').val() === 'Custom Namespace')
+            ns = $('#deploy-app-namespace-field-custom').val().trim();
+
+        var has_errors = false;
+        if (jsondata === '') {
+            $('#deploy-app-manifest-file-empty').show();
+            $('#deploy-app-manifest-file').parent().addClass('has-error');
+            has_errors = true;
+        }
+        if (ns.trim() === '' || ns.trim() === 'Enter Namespace Here') {
+            $('#deploy-app-namespace-field-note').show();
+            $('#deploy-app-namespace-field-custom').parent().addClass('has-error');
+            has_errors = true;
+        }
+        if (!has_errors)
+            appdeployer.manager.deploy_app(ns, jsondata);
+    }
+
+    function deploy_dialog_remove_errors() {
+        $('#deploy-app-manifest-file-note').hide();
+        $('#deploy-app-manifest-file-note-details').hide();
+        $('#deploy-app-namespace-field-note').hide();
+        $('#deploy-app-namespace-field-note-details').hide();
+        $('#deploy-app-general-error').hide();
+        $('#deploy-app-manifest-file-empty').hide();
+        $('#deploy-app-namespace-field-custom-empty').hide();
```

Comment:
Suggest giving these all a certain class so they can be hid in one swoop. ```deploy-dialog-aids``` or something like that.